tests/python/test_dict_methods.py

```diff
@@ -555,3 +555,19 @@ def test_next_operator():
     assert (True)
   fourth = next(myit, 'default')
   assert fourth == 'default'
+
+
+def test_next_operator_non_iterator():
+  make_js_generator = pm.eval("""
+  function* sliceGenerator(pyIter)
+  {
+      yield python.eval('lambda x: next(x)')(pyIter);
+  }
+  sliceGenerator;
+  """)
+
+  try:
+    next(make_js_generator(range(0,5)))
+    assert (False)
+  except StopIteration as e:
+    assert (True)
```

Comment:
it should not be a `StopIteration` exception, it should be the underlying reason, which is a `TypeError` (`'range' object is not an interator`), potentially wrapped in a `SpiderMonkeyError` since the exception occured with JS code on the stack (though I think if we were to improve our error handling, it should just be a `TypeError` with the JS code in the stack trace, but thats outside the scope of this issue)